wiki/osu!_File_Formats/Osu_(file_format)/en.md

```diff
@@ -176,90 +179,206 @@ Example of an inherited Timing Point:
+
+*curvePoints (x:y|...)* is a series of `|`-separated coordinates describing the control points of the slider.
+
+A **linear** slider has a start, specified in *x* and *y* from the common fields, and an end point specified in *curvePoints*. For example `L|100:200`.
+
+A **perfect** circle slider is defined by three points. In that order: start, pass-through, and end. *x* and *y* define the start point, and *curvePoints* defines the pass-through and end point. For example `P|250:200|200:15`.
+
+A perfect circle slider could be represented as a center point, a radius, and two angles for convenience. See this source code for the conversion algorithm: [CircularArcApproximator.cs](https://github.com/ppy/osu/blob/master/osu.Game/Rulesets/Objects/CircularArcApproximator.cs).
+
+A **Bézier** slider is made of one or many Bézier curves, sharing common ends. The degree of each curve is arbitrary. The first control points is defined with *x* and *y*, and the other ones by *curvePoints*.
 
-`424,96,66,2,0,B|380:120|332:96|332:96|304:124,1,130,2|0,0:0|0:0,0:0:0:0:`
+To identify the separation between two curves, the intersection point is repeated. Consider the sequence ABCDDEFFG. You would get the 3 Bézier curves: ABCD (cubic), DEF (quadratic) , FG (linear).
 
-*x*, *y*, *time*, and *type* behave the same as described in Hit Circle Syntax.
+Example: `476,340,6419,2,0,B|437:336|422:309|422:309|384:309|359:337|359:337|328:308|300:304|300:304|272:352|237:383|176:356|159:287|224:256,1,420,6|2,0:0|0:0,0:0:0:0:`
 
-*hitSound* applies to the body of the slider, but only whistle sounds will play during the slider body.
+The first Bézier curve in the segment is quadratic with the following points: (476, 340), (437, 336), (422, 309).
 
-*sliderType* will be `L` (linear), `P` (perfect), `B` (Bezier), or `C` (Catmull). A slider created in the editor with only a start and end point will be a linear slider. A slider with only its start, end, and one grey point will be a perfect circle slider. All others will be Bezier. Catmull sliders are deprecated.
+#### Repeat
 
-*curvePoints (x:y|...)* is a series of `|`-separated coordinates describing the control points of the slider. Red points appear twice. NOTE: curvePoints is separated from sliderType with a `|`, not a comma.
+*repeat* (Integer) is the number of times a player will go over the slider. A value of 1 will not repeat, 2 will repeat once, 3 twice, and so on.
 
-*repeat (Integer)* is the number of times a player will go over the slider. A value of 1 will not repeat, 2 will repeat once, 3 twice, and so on.
+#### Length and duration
 
-*pixelLength (Float)* is the length of the slider along the path of the described curve. If the length is greater than that of the described curve, the slider will continue in a straight line.
+*pixelLength* (Float) is the length of the slider along the path of the described curve. It is specified in osu!pixels, i.e. relative to the 512×384 virtual screen.
```

Comment:
yeah, it's a `Double` here

Reply:
I believe there's the *format*, and the *parser implementation*.

For a format specification, what matters is the notation. Here a *floating number* is implicitly something like `[0-9]*(.[0-9]*)?`. I could write a thousand decimals of pi, it would not matter and it would *work* because we're talking about a textual representation, not a binary representation. Conversely, the value 0.1 cannot be expressed exactly as a Double, technically.

The notion of *Double* is only relevant to the C# implementation, and doesn't even matter in most cases because 32-bit is already super precise for a number which is most of the time integral. High-level languages like Python or JavaScript don't even have the word double, because their floats are already 64-bit.

Even from the point of view of a mapper with no programming experience, the word *double* is going to be confusing I think.

Reply:
by that logic I'd just write "Decimal" then

Reply:
Sold!